fix Issue 17275 - [REG 2.072.0] AssertError@declaration.d(2121): Asse…

WalterBright opened the pull request:
…rtion failure

LemonBoy committed:
Fix issue 6227 - Deprecate comparison of different enums

LemonBoy committed:
Add changelog entry for bugzilla 6627

jacob-carlborg committed:
Refactor Objective-C frontend integration

The major reason for this refactoring is to move the decision if the
Objective-C integration is enabled or not from a compile time
(when building the compiler) decision using the makefile to a runtime
(when running the compiler) decision. This allows to remove the extra
objc_stubs.d file and is required for a cross-compilers like LDC, GDC
and when DMD is cross-compiling for x86 64bit <-> 32bit.

MartinNowak committed:
fix Issue 17338 - [Reg 2.075] link failure unsupported symbol section

- fix usage of "guessed" comdat section index (section_cnt + 1), which
  ignored extended section indices with more than 65280 sections
- fixed by setting the weak identifying group section symbol (sh_info)
  after creating a .text section for it

WalterBright committed:
Merge pull request #6717 from MartinNowak/fix17338

fix Issue 17338 - [Reg 2.075] link failure unsupported symbol section

MartinNowak committed:
Merge pull request #6685 from MartinNowak/toolchain_info

print toolchain info in auto-tester

MartinNowak committed:
add regression test for Issue 17338

MartinNowak committed:
workaround Issue 17339 - ambiguous mangling with const alias args

- with multiple definitions with identical mangling getsegment(".text", p)
  returns the section from the previous instance and later the linker complains
  about a group referencing an earlier section
- fixed by reusing the existing group section (while leaving the new one empty)

WalterBright committed:
Merge pull request #6714 from MartinNowak/fix17337

fix Issue 17337 - SIGILL for AVX vector initialization

WalterBright committed:
Merge pull request #6718 from MartinNowak/fix17338

fixup for - #6717 [Reg 2.075] link failure unsupported symbol section

MartinNowak committed:
fix Issue 17339 - ambiguous mangling with const alias argument

- don't constfold const/immutable variables to generate
  the mangling for alias template arguments

MartinNowak committed:
remove Issue 17339 workaround

davispuh committed:
Fix ASTNull

Parser!ASTNull(null, null, false)

WalterBright committed:
fix Issue 17289 - With Xcode 8.3 linker, warnings of 'pointer not aligned'

MartinNowak committed:
Merge pull request #6721 from WalterBright/fix17289

fix Issue 17289 - With Xcode 8.3 linker, warnings of 'pointer not ali…

MartinNowak committed:
Merge pull request #6720 from davispuh/astnull

Fix ASTNull

MartinNowak committed:
move supported vector ops logic to Target

- because it obviously depends on backend and target architecture

MartinNowak committed:
fix Issue 17344 -  ICE with assignment of post inc-/decremented integral vector

MartinNowak committed:
add SSE4.1 int/uint[4] multiplication (PMULLD)

WalterBright committed:
Merge pull request #6719 from MartinNowak/fix17339

fix Issue 17339 - ambiguous mangling with const alias argument

WalterBright committed:
Merge pull request #6722 from MartinNowak/fix17344

fix Issue 17344 -  ICE with assignment of post inc-/decremented integral vector

WalterBright committed:
more switching to PASSsemanticdone

WalterBright committed:
expand cppmangle.d testing

MartinNowak committed:
Merge pull request #6726 from WalterBright/testcppmangle

expand cppmangle.d testing

MartinNowak committed:
Merge pull request #6725 from WalterBright/as-semanticdone

more switching to PASSsemanticdone

WalterBright committed:
incompatibleTypes() should propagate ErrorExp

WalterBright committed:
refactor: checkNestedRef() to hasStaticLinkTo()

WalterBright committed:
Merge pull request #6679 from jacob-carlborg/objc_refactor

Refactor Objective-C frontend integration

MartinNowak committed:
Merge pull request #6728 from WalterBright/incompatibleTypes

incompatibleTypes() should propagate ErrorExp

MartinNowak committed:
change to isVectorTypeSupported

- better logic than negated check* functions

MartinNowak committed:
change checkVectorType to isVectorTypeSupported

UplinkCoder committed:
[trivial] correct error messages in idgen.d

WalterBright committed:
Merge pull request #6723 from MartinNowak/vector_ops_target

move supported vector ops logic to Target

yebblies committed:
Merge pull request #6730 from UplinkCoder/correctIdgne

[trivial] correct error messages in idgen.d

marler8997 committed:
Moved validation of "tochars" array in tokens.d from runtime to compile time.

yebblies committed:
Merge pull request #6733 from marler8997/ctfetokens

Moved "tochars" validation from RUNTIME to COMPILE-TIME

WalterBright committed:
Merge pull request #6729 from WalterBright/hasStaticLinkTo

refactor: checkNestedRef() to ensureStaticLinkTo()

rainers committed:
add new file staticcond.d to VS project

WalterBright committed:
Merge pull request #6740 from rainers/fix_vsbuild11

add new file staticcond.d to VS project

rainers committed:
fix issue 17367 - force proper alignment for elements in an enumerator field list

Commit:
Merge pull request #4766 from ibuclaw/nogcposix

Merge DWARF symbol info into CVDWARF, don't set language as DW_LANG_C89
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

wilzbach committed:
Fix coverage reports with the new src/ddmd layout

Commit:
Merge pull request #6742 from wilzbach/fix-codecov

Fix coverage reports with the new src/ddmd layout
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

ibuclaw committed:
Remove SH64 predefined version condition.

Commit:
Merge pull request #6744 from ibuclaw/rm-sh64

Remove SH64 predefined version condition.
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

yebblies committed:
Let Target determine how to mangle extern(C++) symbols

Commit:
Merge pull request #6745 from yebblies/cppmanglert

Let Target determine how to mangle extern(C++) symbols
merged-on-behalf-of: Iain Buclaw <ibuclaw@gdcproject.org>

arBmind committed:
Added bugtracker keyword links

* added categories `trivial`, `bootcamp` and `preapproved` as direct links to encourage a good start for new contributors.

RazvanN7 committed:
Fix Issue 7016

MartinNowak committed:
change verstr.h to VERSION

- no longer need the C string quotes

arBmind committed:
fixed visual studio build to generate VERSION instead of verstr.h

MartinNowak committed:
Merge pull request #1 from arBmind/version_file

fixed visual studio build to generate VERSION instead of verstr.h

MartinNowak committed:
typos

Commit:
Destructors can be called on shared objects

arBmind committed:
fixed ImportPath to be consistent on all build variants

Commit:
Merge pull request #6747 from arBmind/patch-1

Added bugtracker keyword links
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

ibuclaw committed:
Move logic for determining extern(System) to Target

ibuclaw committed:
Use Target.systemLinkage where appropriate.

MartinNowak committed:
fixup! fixed visual studio build to generate VERSION instead of verstr.h

MartinNowak committed:
Merge branch 'version_file' of git://github.com/arBmind/dmd into version_file

Commit:
Merge pull request #6751 from ibuclaw/sys-linkage

Move logic for determining extern(System) to Target
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

kinke committed:
Fix alignment/size of packed aggregates (issue #17277)

Change the definition of an aggregate's overall alignment from maximum of
each field's `max(natural type alignment, optional alignment attribute)`
to maximum of each field's actual alignment (which may be less than its
type's natural alignment).

MartinNowak committed:
Merge pull request #6749 from MartinNowak/version_file

replace verstr.h with VERSION

yebblies committed:
Revert "Struct array == lowering to object.__equals (+changelog)"

This reverts commit 34dd5c9ff14151af1d5c36101866513f241d235e.

yebblies committed:
Remove toModule* and m* dead code

Commit:
Merge pull request #6757 from yebblies/revert6679

Revert "Struct array == lowering to object.__equals (+changelog)"
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

Commit:
Merge pull request #6758 from yebblies/toModule

Remove `toModule*` and `m*` dead code
merged-on-behalf-of: Iain Buclaw <ibuclaw@gdcproject.org>

yebblies committed:
Remove imports and headers that are unneeded now that the glue layer is in D

UplinkCoder committed:
improve hdrgen to print valid d-code on unrolled loop statements and make fullDump print all template members

Commit:
Merge pull request #6761 from yebblies/importmars

Remove imports and headers that are unneeded now that the glue layer is in D
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
fix Issue 17356 - [Reg 2.075] __simd_sto no longer executed

ibuclaw committed:
Add LINKsystem, remove dependency on Target from the parser

ibuclaw committed:
Add tests for extern(System) linkage errors

WalterBright committed:
Merge pull request #6741 from rainers/issue_17367

fix issue 17367 - CodeView/MSCOFF: bad debug information for enumerators

WalterBright committed:
refactor: convert scodelem() to CodeBuilder

Commit:
Merge pull request #6765 from WalterBright/cdb-scodelem

refactor: convert scodelem() to CodeBuilder
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
refactor: convert codelem() to CodeBuilder

WalterBright committed:
Merge pull request #6764 from ibuclaw/linksystem

Add LINKsystem, remove dependency on Target from the parser

Commit:
Merge pull request #6768 from WalterBright/cdb-codelem

refactor: convert codelem() to CodeBuilder
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
refactor: convert cdxxx[] to CodeBuilder

WalterBright committed:
fix Issue 13331 - naked asm functions are broken when compiling with -profile

sibnick committed:
NetBSD changes

WalterBright committed:
refactor: setArray() in e2ir.d

WalterBright committed:
Merge pull request #6772 from nrTQgc/NetBSD

NetBSD changes

WalterBright committed:
fix Issue 16197 - Constructors/postblits and destructors don't match up for array initialisation

WalterBright committed:
convert Bugzilla to URL

Commit:
Merge pull request #6775 from WalterBright/Bugzilla

convert Bugzilla to URL
merged-on-behalf-of: Iain Buclaw <ibuclaw@gdcproject.org>

WalterBright committed:
fix Issue 16540 - Attributes do not propagate correctly in lazy params

Commit:
Merge pull request #6770 from WalterBright/fix13331

fix Issue 13331 - naked asm functions are broken when compiling with …
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

WalterBright committed:
fix Issue 17396 - Add colorized syntax highlighting to error messages emitted by dmd

WalterBright committed:
traits.d: fix error messages to use syntax highlighting

andralex committed:
Merge pull request #6777 from WalterBright/fix17396

fix Issue 17396 - Add colorized syntax highlighting to error messages…

WalterBright committed:
statement.d: fix error messages to use syntax highlighting

WalterBright committed:
lexer.d: fix error messages to use syntax highlighting

WalterBright committed:
fix error recovery in color syntax highlighting

WalterBright committed:
refactor: split off color handling into console.d

WalterBright committed:
toobj.d: fix error messages to use syntax highlighting

WalterBright committed:
toir.d: fix error messages to use syntax highlighting

WalterBright committed:
todt.d: fix error messages to use syntax highlighting

Commit:
Merge pull request #6781 from WalterBright/fix-color

fix error recovery in color syntax highlighting
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

WalterBright committed:
Merge pull request #6778 from WalterBright/traits-errors

traits.d: fix error messages to use syntax highlighting

WalterBright committed:
staticassert.d: fix error messages to use syntax highlighting

Commit:
Merge pull request #6779 from WalterBright/statements-errors

statement.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

Commit:
Merge pull request #6784 from WalterBright/toir-errors

toir.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

Commit:
Merge pull request #6783 from WalterBright/toobj-errors

toobj.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

WalterBright committed:
staticcond.d: fix error messages to use syntax highlighting

WalterBright committed:
access.d: fix error messages to use syntax highlighting

dlang-bot committed:
Merge pull request #6780 from WalterBright/lexer-errors

lexer.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

WalterBright committed:
aggregate.d: fix error messages to use syntax highlighting

WalterBright committed:
aliasthis.d: fix error messages to use syntax highlighting

WalterBright committed:
statementsem.d: fix error messages to use syntax highlighting

dlang-bot committed:
Merge pull request #6791 from WalterBright/aliasthis-errors

aliasthis.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

dlang-bot committed:
Merge pull request #6788 from WalterBright/access-errors

access.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

dlang-bot committed:
Merge pull request #6790 from WalterBright/aggregate-errors

aggregate.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

CyberShadow committed:
Convert line endings from DOS to UNIX

dlang-bot committed:
Merge pull request #6787 from WalterBright/staticcond-errors

staticcond.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

dlang-bot committed:
Merge pull request #6785 from WalterBright/todt-errors

todt.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

WalterBright committed:
Merge pull request #6793 from CyberShadow/pull-20170515-054941

Convert line endings from DOS to UNIX

dlang-bot committed:
Merge pull request #6792 from WalterBright/statementsem-errors

statementsem.d: fix error messages to use syntax highlighting
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

dlang-bot committed:
Merge pull request #6782 from WalterBright/add-console.d

refactor: split off color handling into console.d
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

WalterBright committed:
refactor: improve console.d encapsulation
1. encapsulated state in a struct Console
2. eliminated random global variables
3. removed hardcoding of stderr
4. added documentation
5. removed grossly excessive number of system calls (on Windows version)
6. provide reasonable fallback if console fails to initialize
7. reduce messy interleaved version declarations
8. remove ugly C++ conversion detritus
9. fix incorrect error checking of GetConsoleScreenBufferInfo()
10. use fputs instead of fprintf where possible
11. put initialization of console after command line processing, not before. Doing it before could mean it was turned on
even if it failed.

MartinNowak committed:
remove debug printf

dlang-bot committed:
Merge pull request #6763 from WalterBright/fix17356

fix Issue 17356 - [Reg 2.075] __simd_sto no longer executed
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

WalterBright committed:
Merge pull request #6786 from WalterBright/staticassert-errors

staticassert.d: fix error messages to use syntax highlighting

dlang-bot committed:
Merge pull request #6794 from WalterBright/refactor-console

refactor: improve console.d encapsulation
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

ibuclaw committed:
Fix out of bounds bug caught by compiler

ibuclaw committed:
Fix SEGV in evalu8 using G++ version 7.

dlang-bot committed:
Merge pull request #6769 from WalterBright/cdb-cdxxx

refactor: convert cdxxx[] to CodeBuilder
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

WalterBright committed:
cod1.c: remove CDXXX macros

WalterBright committed:
cod3.c: remove CDXXX macros

WalterBright committed:
nteh.c: remove CDXXX macros

WalterBright committed:
cgxmm.c: remove CDXXX macros

WalterBright committed:
cg87.c: remove CDXXX macros

WalterBright committed:
cod4.c: remove CDXXX macros

WalterBright committed:
cod2.c: remove CDXXX macros

WalterBright committed:
Merge pull request #6801 from WalterBright/cod4-CodeBuilder

cod4.c: remove CDXXX macros

WalterBright committed:
Merge pull request #6800 from WalterBright/cg87-CodeBuilder

cg87.c: remove CDXXX macros

WalterBright committed:
Merge pull request #6799 from WalterBright/cgxmm-CodeBuilder

cgxmm.c: remove CDXXX macros

WalterBright committed:
Merge pull request #6798 from WalterBright/nteh-CodeBuilder

nteh.c: remove CDXXX macros

WalterBright committed:
Merge pull request #6796 from WalterBright/cod1-CodeBuilder

cod1.c: remove CDXXX macros

WalterBright committed:
Merge pull request #6797 from WalterBright/cod3-CodeBuilder

cod3.c: remove CDXXX macros

UplinkCoder committed:
set refcount to one to avoid immediate freeing

WalterBright committed:
Merge pull request #6752 from atilaneves/fix_shared_dtor

Fix compiler-inserted destructor call on shared objects

WalterBright committed:
Merge pull request #6795 from ibuclaw/fixgxx7

Fix bugs when building with g++-7

Commit:
Fix make clean to remove stale backend/cdxxx.c and other optabgen files.

WalterBright committed:
Merge pull request #6804 from quickfur/cdxxx

Fix make clean to remove stale backend/cdxxx.c .

dlang-bot committed:
Merge pull request #6802 from WalterBright/cod2-CodeBuilder

cod2.c: remove CDXXX macros
merged-on-behalf-of: unknown

WalterBright committed:
remove CDXXX macros from code.h

WalterBright committed:
syntax highlighting for IdentifierExp

dlang-bot committed:
Merge pull request #6762 from UplinkCoder/print_mixin_fullAST

make -vcg-ast print all instantiated template
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

dlang-bot committed:
Merge pull request #6809 from WalterBright/errors-identifier

syntax highlighting for IdentifierExp
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

WalterBright committed:
add Classfication enum

WalterBright committed:
refactor cg87.c to use CodeBuilder interfaces

dlang-bot committed:
Merge pull request #6748 from RazvanN7/Fix_Issue_7016

Fix Issue 7016 - local import does not create -deps dependency
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

UplinkCoder committed:
improve mangler error message

WalterBright committed:
fix Issue 17399 - core.checkedint.addu cannot inline function

yebblies committed:
Remove ASTNull

After reviewing this in more detail the conclusion is that there is no use case for the ASTNull AST family that is not better covered by ASTBase and therefore should be removed.

While this does cover the API used by the parser to construct an AST it does not actually result in building an AST, and at the same time is ineffective for use cases that don't require an AST as it will allocate huge amounts of memory and does not provide a hookable API.  The testing of the parser API will be adequately covered by ASTBase.

dlang-bot committed:
Merge pull request #6776 from WalterBright/test16540

fix Issue 16540 - Attributes do not propagate correctly in lazy params
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

CyberShadow committed:
test: Add lexer CRLF tests

dlang-bot committed:
Merge pull request #6808 from WalterBright/rmCDXXX

remove CDXXX macros from code.h
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6814 from UplinkCoder/cpp_mangle_error_message

improve c++ mangle error message
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6812 from WalterBright/cdb-cg87.c

refactor cg87.c to use CodeBuilder interfaces
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
replace hack with isArrayOp

WalterBright committed:
Merge pull request #6817 from CyberShadow/pull-20170521-070343

test: Add lexer CRLF tests

WalterBright committed:
Merge pull request #6818 from WalterBright/isArrayOp

replace hack with isArrayOp

WalterBright committed:
cgxmm.c: switch to CodeBuilder interface

WalterBright committed:
add backticks to dcast.d error messages

dlang-bot committed:
Merge pull request #6820 from WalterBright/syntax-dcast

add backticks to dcast.d error messages
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6773 from WalterBright/setArray

refactor: setArray() in e2ir.d
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6774 from WalterBright/fix16197

fix Issue 16197 - Constructors/postblits and destructors don't match …
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

yebblies committed:
Merge pull request #6819 from WalterBright/cdb-cgxmm.c

cgxmm.c: switch to CodeBuilder interface

dlang-bot committed:
Merge pull request #6759 from yebblies/astnull

Remove ASTNull
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

UplinkCoder committed:
fix @LemonBoy's PR

dlang-bot committed:
Merge pull request #6810 from WalterBright/Classification

add Classification enum
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
fix Issue 17419 - add __traits(getLinkage, s) to the the linkage of symbol s

andralex committed:
Merge pull request #6822 from WalterBright/fix17419

fix Issue 17419 - add __traits(getLinkage, s) to the the linkage of s…

dlang-bot committed:
Merge pull request #6821 from UplinkCoder/b6227

Fix issue 6227 - Deprecate comparison of different enums (revamp)
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

WalterBright committed:
add types to __traits(getLinkage,t)

andralex committed:
Merge pull request #6823 from WalterBright/getLinkage2

add types to __traits(getLinkage,t)

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

WalterBright committed:
add __traits(getFunctionVariadicStyle, f)

WalterBright committed:
Merge pull request #6825 from WalterBright/fix17421

add __traits(getFunctionVariadicStyle, f)

MartinNowak committed:
Merge pull request #6824 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

UplinkCoder committed:
remove {.sort | .reverse} array properties and adjust tests

MartinNowak committed:
remove unused rundtor

- dtor gate var no longer used

WalterBright committed:
fix Issue 17425 - add __traits(getParameterStorageClasses, f, i)

dlang-bot committed:
Merge pull request #6828 from MartinNowak/rm_rundtor

remove unused rundtor
merged-on-behalf-of: Iain Buclaw <ibuclaw@gdcproject.org>

rainers committed:
VS project: add missing file console.d

dlang-bot committed:
Merge pull request #6830 from rainers/fix_vcbuild12

VS project: add missing file console.d
merged-on-behalf-of: Walter Bright <WalterBright@users.noreply.github.com>

UplinkCoder committed:
now set we set the ref there is no need new the buffer

dlang-bot committed:
Merge pull request #6832 from UplinkCoder/fix_vcg_ast

now set we set the ref there is no need new the buffer
merged-on-behalf-of: Iain Buclaw <ibuclaw@gdcproject.org>

WalterBright committed:
Merge pull request #6829 from WalterBright/fix17425

fix Issue 17425 - add __traits(getParameterStorageClasses, f, i)

WalterBright committed:
refactor: us CodeBuilder interface in nteh.c

RazvanN7 committed:
Create parser as a library

dlang-bot committed:
Merge pull request #6754 from kinke/fix17277

Fix alignment/size of packed aggregates (issue #17277)
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6834 from WalterBright/cdb-nteh.c

refactor: use CodeBuilder interface in nteh.c
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
use CodeBuilder interface in cod2.c

dlang-bot committed:
Merge pull request #6838 from WalterBright/cdb-cod2.c

use CodeBuilder interface in cod2.c
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
fix Issue 17450 - escaping delegate context pointer not detected for member functions

WalterBright committed:
cod1.c: more CodeBuilder interfacing

dlang-bot committed:
Merge pull request #6841 from WalterBright/cod1-cdb2

cod1.c: more CodeBuilder interfacing
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6839 from WalterBright/fix17450

fix Issue 17450 - escaping delegate context pointer not detected for …
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
better error message for returning escapes

WalterBright committed:
cod1.c: change interfaces to use CodeBuilder

dlang-bot committed:
Merge pull request #6842 from WalterBright/ret-escape-msg

better error message for returning escapes
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
fix Issue 17451 - ICE in ddmd/declaration.d(2179)

dlang-bot committed:
Merge pull request #6843 from WalterBright/cdb3-cod1.c

cod1.c: change interfaces to use CodeBuilder
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
fix Issue 17275 - [REG 2.072.0] AssertError@declaration.d(2121): Assertion failure

ZombineDev commented on test/fail_compilation/test17275.d:
```diff
@@ -0,0 +1,19 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/test17275.d(12): Error: undefined identifier 'ModuleGroup', did you mean function 'moduleGroup'?
+fail_compilation/test17275.d(12): Error: inout on return means inout must be on a parameter as well for inout(ModuleGroup)()
```
There's already test/fail_compilation/fail17275.d (https://github.com/dlang/dmd/pull/6690/files#diff-278b7365e40009be458ef32b7f8d85a3). Why don't you put the test output in there?

ibuclaw reviewed (changes requested):
As per comment (I actually made such a commit in #6746 here https://github.com/ibuclaw/dmd/commit/407d2a5c3d068243c2899ef95f079170baa91178)

WalterBright committed:
cgcod.c: change interfaces to use CodeBuilder

dlang-bot committed:
Merge pull request #6846 from WalterBright/cdb2-cgcod.c

refactor cgcod.c: change interfaces to use CodeBuilder
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright commented:
@ibuclaw I don't understand what you're asking for.

dlang-bot committed:
Merge pull request #6844 from WalterBright/fix17451

fix Issue 17451 - ICE in ddmd/declaration.d(2179)
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

WalterBright committed:
CodeBuilder: getregs()

WalterBright committed:
generate fewer __inlineretval temps

dlang-bot committed:
Merge pull request #6848 from WalterBright/cdb-getregs

CodeBuilder: getregs()
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
CodeBuilder: allocreg()

WalterBright committed:
refactor: add printast.d

dlang-bot committed:
Merge pull request #6850 from WalterBright/cdb-allocreg

CodeBuilder: allocreg()
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6851 from WalterBright/printast

refactor: add printast.d
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

WalterBright committed:
replace construct-copy with a construct directly into target

WalterBright committed:
CodeBuilder: movregconst()

WalterBright committed:
update readme.txt with Boost license

dlang-bot committed:
Merge pull request #6853 from WalterBright/cdb-movregconst

CodeBuilder: movregconst()
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6849 from WalterBright/inline-tmp

generate fewer __inlineretval temps
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dnadlinger committed:
runnable/template1: Do not hardcode long type alignment

The alignment of D's long type is not what the test is supposed
to verify, so make it less brittle (relevant e.g. for AArch64).

dlang-bot committed:
Merge pull request #6854 from WalterBright/readme-license

update readme.txt with Boost license
merged-on-behalf-of: David Nadlinger <code@klickverbot.at>

dnadlinger committed:
test42: Don't run (1 / real.epsilon) -> double -> ulong test if too large to fit

This is the case with quadruple-precision reals, e.g. on ARM64/AAPCS64.

WalterBright committed:
CodeBuilder: outblkexitcode()

wilzbach committed:
Make Boost license visible

WalterBright committed:
Merge pull request #6861 from wilzbach/make-license-visible

Make Boost license visible

WalterBright committed:
Merge pull request #6858 from klickverbot/skip-epsilon-quad

test42: Don't run (1 / real.epsilon) -> double -> ulong test if too large to fit

RazvanN7 committed:
Add examples directory

RazvanN7 committed:
Add windows targets + fix posix targets

rainers committed:
VC projects: add new file printast.d

WalterBright committed:
Merge pull request #6857 from klickverbot/template-brittle-test

runnable/template1: Do not hardcode long type alignment

rainers committed:
fix Issue 17432 - scope delegates change type, but not mangling

do not add "scope" to .mangleof or .stringof if it was inferred

UplinkCoder committed:
add dd file

MartinNowak committed:
More information in changelog entry

dlang-bot committed:
Merge pull request #6827 from UplinkCoder/remove_arrayProps

remove .sort / .reverse properties and adjust tests
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

WalterBright committed:
Merge pull request #6864 from rainers/issue_17432_2

fix Issue 17432 - scope delegates change type, but not mangling

wilzbach committed:
Unify comment header and expose module with Ddox

WalterBright committed:
Merge pull request #6867 from wilzbach/ddoxify-dstruct

Unify comment header and expose module with Ddox

RazvanN7 committed:
Fix posix.mak to compile root files once

dlang-bot committed:
Merge pull request #6863 from rainers/fix_vcbuild13

VC projects: add new file printast.d
merged-on-behalf-of: Vladimir Panteleev <github@thecybershadow.net>

RazvanN7 committed:
Address review requests

RazvanN7 committed:
Fix win32.mak to match posix hierarchy

dlang-bot committed:
Merge pull request #6836 from RazvanN7/ASTBase

Create parser as a library
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

dlang-bot committed:
Merge pull request #6873 from RazvanN7/WindowsMakefile

Fix win32.mak to match posix hierarchy
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

dlang-bot committed:
Merge pull request #6860 from WalterBright/cdb-outblkexitcode

CodeBuilder: outblkexitcode()
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

dlang-bot committed:
Merge pull request #6815 from WalterBright/fix17399

fix Issue 17399 - core.checkedint.addu cannot inline function
merged-on-behalf-of: unknown

dlang-bot committed:
Merge pull request #6852 from WalterBright/copy-elision

replace construct-copy with a construct directly into target
merged-on-behalf-of: Daniel Murphy <yebblies@gmail.com>

ibuclaw commented:
@WalterBright - you don't need a new test case, you just need to copy the `TEST_OUTPUT` to  test/fail_compilation/fail17275.d

WalterBright commented:
> @WalterBright - you don't need a new test case, you just need to copy the TEST_OUTPUT to test/fail_compilation/fail17275.d

Well, fail17275 was in another branch, and now I've fouled it up completely trying to merge the two.

ibuclaw commented:
How did it end up in another branch?
